istream.h

```diff
@@ -75,8 +75,8 @@ struct ssl_stream {
     struct ssl_handle *handle;
     char type;
     char iseos;
-    int (*read) ();
-    void (*close) ();
+    int (*read) (void *, unsigned char *, int);
+    void (*close) (void *, int, int *);
```

Comment:
These functions are a bit problematic because they can either be stdio functions or unix functions that operat on fd. Specifying `FILE *` gives the wrong impression that they're meant for stdio only.

Personally I'd go with either void parameter as the type `f(void)` or just a regular `void *` and then cast it to proper type on usage. A union might also be a possibility.